doc(book): Fill up empty sections and add table diagram

Nashtare opened the pull request:
Small tweaks:

- fix bibliography quotes display
- point to README for book cover
- add table intro + diagram

0xaatif reviewed (approved):
Love mermaid :))

I do wonder if there's any way for us to not check in 1000 lines of js into the repo

Nashtare commented:
@0xaatif I was thinking of including by default as this is the result of `mdbook-mermaid install book` (i.e. initialization step) but we could have it in `.gitignore` and add the step to the deployment workflow if you prefer

0xaatif commented:
That's my preference if it's easy, but not a blocker :)

Nashtare commented:
Yeah it's fairly trivial, will update it.

Nashtare commented:
@0xaatif I've added the files to `.gitignore` and added the step in CI.
I've also decoupled the `build` and `deployment` steps so that we always run `build` on PRs, and only `deploy` when merging on `develop` / `main`, figured we'd rather catch any build failures ahead :')

atanmarko reviewed (approved):
I don't really know anything about `mermaid` so I trust you on this one 😀

0xaatif reviewed (approved):
Love it!

Nashtare commented:
Hmm.. the website is returning 404 even though deployment seems to have run properly:
https://github.com/0xPolygonZero/zk_evm/actions/runs/10773324121

Not sure what's the issue here
